File: test/easy/find_words_containing_test.dart
import 'package:test/test.dart';
import 'package:dark_coding_challenges/easy/find_words_containing.dart';

class FindWordsContainingTestCases {
  List<String> words;
  String c;
  List<int> expected;
  FindWordsContainingTestCases(this.words, this.c, this.expected);
}

void main() {
  group("Feature:", () {
    final testCases = [
      FindWordsContainingTestCases(["abc", "bcd", "aaaa", "cbc"], "a", [0, 2]),
      FindWordsContainingTestCases(["leet", "code"], "e", [0, 1]),
      FindWordsContainingTestCases(["abc", "bcd", "aaaa", "cbc"], "z", []),
    ];

    for (final step in testCases) {
      final words = step.words;
      final c = step.c;
      final expected = step.expected;
      test("", () {
        List<int> actual = findWordsContaining(words, c);
        expect(actual, equals(expected));
      });
    }
  });
}

File: test/easy/find-words-containing.test.ts
import { describe, test, expect } from "bun:test";
import { findWordsContaining } from "@/src/easy/find-words-containing";
describe("findWordsContaining", () => {
    test.each([
        [["abc", "bcd", "aaaa", "cbc"], "a", [0, 2]],
        [["leet", "code"], "e", [0, 1]],
        [["abc", "bcd", "aaaa", "cbc"], "z", []],
    ])("", (words: string[], char: string, output: number[]) => {
        const result = findWordsContaining(words, char);
        expect(result).toStrictEqual(output);
    });
});
